api/src/wfl/server.clj

```diff
@@ -154,14 +159,15 @@
   (loop []
     (if-let [f (first (filter future-done? futures))]
       (do @f nil)
-      (do (.sleep TimeUnit/SECONDS 20)
+      (do (util/sleep-seconds 20)
           (recur)))))
 
 (defn run
-  "Run server in ENVIRONMENT on PORT."
+  "Run server in `environment` on `port`."
   [& args]
   (log/info (str/join " " ["Run:" wfl/the-name "server" args]))
   (let [port    (util/is-non-negative! (first args))
         manager (start-workload-manager)
-        logger  (start-logging-polling)]
-    (await-some manager logger (start-webserver port))))
+        logger  (start-logging-polling)
+        slacker (slack/start-notification-loop slack/notifier)]
```

Comment:
I wonder if this had any other bad side effects.
I can't think of any, but it was definitely a bug.

Reply:
I think it would be nice to have logs behave more like the `wfl.debug/trace` macro.
That way you would see a query and its result, or an API call and its response together.
And you wouldn't have to wrap calls to `log/whatevz` in a `do` or `when` and so on.

Reply:
> I wonder if this had any other bad side effects.

I wonder if those times where the update loop was "stuck" with no logs could have been driven by an issue in the discarded `manager` future which didn't propagate up.

Should we define the `slacker` above the `manager` since we Slack within the workload update loop?

Reply:
It shouldn't matter, but makes sense.

Reply:
Yeah. It better not matter, so I leave it as is.